### PR TITLE
Added Android implementation

### PR DESCRIPTION
**Pull request by isnifer**
Basic Android implementation of Tipsi Barcode Scanner

### CODE REVIEW

**Comment by fAns1k** [comment]
https://github.com/tipsi/tipsi-barcode/blob/feature/android-implementation/android/src/main/java/com/gettipsi/barcode/TPSBarcodeManager.java#L47

This line is redundant, 'cause onHostPause already call before onHostDestroy, and now method stop() called twice.

**Comment by isnifer** [comment]
@fAns1k fixed

File: android/src/main/java/com/gettipsi/barcode/TPSBarcodeView.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        int barcodeFormats = Barcode.ALL_FORMATS;
+        createCameraSource(autoFocus, useFlash, barcodeFormats);
+        startCameraSource();
+    }
+    /**
+     * Override in activity to handle barcode detector updates
+     *
+     * @param barcodes - a list of detected barcodes
+     */
+    protected void onBarcodesUpdated(List<Barcode> barcodes){
```
Comment (fAns1k) [inline android/src/main/java/com/gettipsi/barcode/TPSBarcodeView.java]: If this class don't have a children, "protected" need replace to "private".

File: android/src/main/java/com/gettipsi/barcode/TPSBarcodeView.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+                .getJSModule(RCTDeviceEventEmitter.class)
+                .emit("scannerBarcodeRead", event);
+        }
+    }
+
+    /**
+     * Removes update listener from graphic overlay
+     */
+
+    protected void stopUpdates(){
```
Comment (fAns1k) [inline android/src/main/java/com/gettipsi/barcode/TPSBarcodeView.java]: If this class don't have a children, "protected" need replace to "private".

**Comment by isnifer** [comment]
@fAns1k fixed

**Comment by fAns1k** [comment]
Looks good to merge.

**Pull request merged**